### ISSUE TITLE
Feedback: Aug 2, 2019 

### ISSUE THREAD

**Issue by brianamarie** [githubtraining/react-course#31]
# React Feedback - Aug 2, 2019

## TL;DR

@partyshah Thank you for continuing to work on and improve this course! I have gone through it again, and made some detailed comments. Please put some time on my calendar if you'd like to go through this synchronously. cc @hectorsector

### 👍

- I REALLY LOVE THIS COURSE! Learning React the first time around was really hard for me, specifically because if I made a mistake early on, I was blocked from connecting the dots and understanding how the big picture worked. This course helps users get past exactly that, because even if they don't understand one earlier step in complete detail, they're able to move forward and see the big picture.
- I'm **so happy** to see more steps! I was able to get through the course with a _much better_ understanding of the concepts. There are places where I think you can and should divide further - I've put detail under each step.
- Course preferences - is there a plan to incorporate the choice to complete this course on .com?
- In general, I like the duplication of the instructions in the issue comments and the template, so when the user gets to the template, they're reminded of what they're supposed to be doing.

### Multiple comments

- I'm against multiple comments whenever possible. I think it adds confusion. Whenever possible, I'd recommend condensing each step to one response/comment (with the exception of a response pointing the user to the next issue, if it's somewhere else).
- I _do see value_ in multiple comments when explaining lines of code. To make everything more clear, make sure there is an ` ### :keyboard: Activity: [activity-name]` called out _only once per step_. But, if there will be code callouts that don't require action, I think that the multiple comments could be useful. I would strongly recommend titling them in a way that makes it clear what's going on to the user - something like "Code call-out 1/3: Passing functions as props", followed by the conceptual information associated with that line of code. I very much prefer this style of calling attention to code vs pasting it into a response, which is also done at times in this course.

### Response format ⌨️

- ️Every response should have a distinct difference between **information** and **activity**. The activity portion should always be called out and formatted with `### :keyboard: Activity: [name-of-activity]`, and the steps should be broken up into a numbered list. ⌨️

### Exact instructions - line numbers 🔢

- There are quite a few places where a user should do something on a specific line. I **love** that the line is called out in the template, but I still made mistakes and got lost. Every time an instruction tells a user to change a line in a file, we should tell them exactly which line and exactly which file. It's better to tell them too many times than not enough times! 🔢

### Assorted

- The [gist link](https://gist-playground.github.com/ea549306dcee3017e04c55200457e4b5) in the [tracking issue](https://github.com/github/learning-engineering/issues/584) didn't work for me. I still make this mistake all of the time - when sharing a gist link for a playground, make sure you use the "share" button on the lower left side instead of the URL you're using.

### Landing page - pre-enrollment

- There's a few wording things, like punctuation (:) that would help this read more smoothly. I think a once-over for grammar, while considering image placement, would be good. Let me know if you aren't sure what I mean and I'll be happy to open a PR.
- The screenshots are helpful.
  - Could you make them a bit tighter, so there's less white space?
  - I think only a screenshot of the final product is necessary for the course landing page.
  - I think the other screenshots would be helpful throughout the course, where the user is supposed to be seeing each step. Then, they can compare what they're seeing with what they're supposed to be seeing.

### Once enrolled

- I'm **so happy** to see more consistent, clear step titles, descriptions, and links. It helps me know what is going to happen at each step, so I can start organizing my understanding immediately.

### Issue 1

- Watch out for abbreviations. I noticed that "PR" is fixed :tada: but I still noticed "repo" instead of "repository" quite a few times.
- "I recommend" - avoid single first person language. Instead, consider passive "x is recommended" or "we recommend".
- :keyboard: This issue needs response format tweaked so it's clearly **content** and **activity**. See above ☝🏻_Response format_
- I'm _so happy_ that there are generally more steps. I think you **need to go further**. I would break this step out differently. 
  1. Activity: Open terminal, CD into desired location, clone repository, CD into repository, check for node, npm install,npm start, then close the issue when they're ready to move on. (What's at risk by assuming people can do those things but they can't? A LOT! If you want proof let's pick a few test-subjects to go through this course while you watch. :wink: What's at risk by telling an expert to CD into the desired location? Very, very little.)
  2. Response: Instead of telling the user to open a pull request, the response would be that Learning Lab opens the first pull request. The concept info for components and instructions for adding a header component would be in the body of the pull request. Learning Lab would also comment on the closed issue with a link to the newly opened PR.
- The issue didn't close when I opened my first pull request, which I would have expected. If you keep the step where the user opens a pull request instead of Learning Lab, make sure the PR being opened also triggers the issue deletion.

### PR 1 - Add a header component

- I'm immediately confused by why there's a first and a second comment separated. I can see the reason for the line comment to be somewhere different, but why not combine all three? Why shouldn't all of the text be in one place, so the user knows exactly where to look, and exactly what they're being told?
- :keyboard: This step needs response format tweaked so it's clearly **content** and **activity**. See above ☝🏻_Response format_
  - Ex: Info = components. Action = add a header component.
- Check the wording here. I'm concerned it may be on the fence of too casual. Ex: "??", "bunch"
- 🔢I first noticed that I didn't know exactly what line I should be changing. The wording was "Replace the line where it says"...what line is that? Being as specific as possible is going to help. See above ☝🏻_Exact instructions - line numbers_

### PR 2 - Child component

- :keyboard: This step needs response format tweaked so it's clearly **content** and **activity**. See above ☝🏻_Response format_. Ex:
  - Info: Props, child component, difference between child component and other components
  - Activity: Import a child component

### PR 3 - Uncomment

- I really like that all of the responses are in one comment, but they still need the delineation of **content** and **activity**. See above ☝🏻_Response format_
- What is _uncommenting_ teaching the user? Is this about the child component that we're importing in the earlier step? The context here should be a little louder, so users don't just follow the directions of uncommenting without missing the picture of what you're trying to teach them.

### PR 4 - Replace title

- What is the importance of title? I think the concept here could be a little louder, so I understand as a user, why does a title matter?
- A few language things - do we need to say "ugly"?
- :keyboard: This step needs response format tweaked so it's clearly **content** and **activity**. See above ☝🏻_Response format_
- 🔢 Here, I also didn't know exactly what line I should be changing. The wording was "Replace the line where it says"...what line is that? Being as specific as possible is going to help. See above ☝🏻_Exact instructions - line numbers_
- I'm noticing that some of the comments in the template (step 4, step 3, etc) don't match up to the new steps. Just a reminder to update those. :blush:

### PR 5 - State variables

- This feels like a sudden shift to so many comments! Why are there 4 comments? If state needs to be taught in multiple comments, why shouldn't it have multiple steps?
- On that same theme, "state" feels very separate from "components". What if the concepts took place, along multiple steps each, in their own PRs? That would help me identify and differentiate the concepts.
- "Set it equal to an empty list" - Do you mean array? Object? How is the user supposed to know that?
- The "create add student function" is a leap in understanding from previous steps. I also did not do it because my JS is rusty. I wasn't caught, or helped.
  - Let's check for if the users are doing this correctly
  - Let's also give them a bit more help - whether it's the first time or after we notice they need help, maybe a "hint" dropdown for someone who wants a function written properly but needs help?
- I also missed a step to uncomment something, overwhelmed a bit by comments. But, the course stopped me! I want it to tell me in more detail when it finds out I missed something. What line? What exact text? What's the concept again, explained in different words?
- As I said before, I would break this step up into at least 3 steps, and maybe in its own PR. 

### Pr 6 - Fix everything

- This has 3 comments again. Do they need to be 3 comments? Where are the actions vs the content?
- Fix "everything" - I literally had no idea that anything was broken. Can this step title be more descriptive, with more context in the opening paragraph, so I know what's broken?
